lib/vnet/admin_process_darwin.go

```diff
@@ -113,7 +117,29 @@ func RunDarwinAdminProcess(ctx context.Context, config daemon.Config) error {
 			}
 		}
 	})
-	return trace.Wrap(g.Wait(), "running VNet admin process")
+
+	done := make(chan error)
+	go func() {
+		done <- g.Wait()
+	}()
+
+	select {
+	case err := <-done:
+		return trace.Wrap(err, "running VNet admin process")
+	case <-ctx.Done():
+		timer := time.NewTimer(5 * time.Second)
+		defer timer.Stop()
+
+		select {
+		case <-done:
+			// network stack exited cleanly within timeout
+			return trace.Wrap(err, "running VNet admin process")
+		case <-timer.C:
```

Comment:
```suggestion
		case <-time.After(10 * time.Second):
```

This is the new way of doing this, see the docs for [`time.After`](https://pkg.go.dev/time#After). I'd also bump the timeout to 10s just to be safe, the error message will have to be updated as well.